### DISCUSSION TITLE
C# switch pattern matching one default.

### DISCUSSION THREAD

**Discussion by Terry1511512** [dotnet/csharplang#3897] [General]
I would like to see the pattern matching on a switch statement extended to the default case.
IE:
switch( SomeValue )
{
case Value1 when ( something == 0)
break;

case Value2 when ( something == 1)
break;

default when ( something == 1)
break;

default when ( something == 3)
break;

default:
break;
}

**Comment by RikkiGibson** [comment]
Consider using `case var _ when (something == 1)`, for example.

I feel like there might be a simpler way to do this, such as `case _ when ...`, but it doesn't seem to be accepted.

**Reply by Terry1511512** [reply]
Only did it the long way for clarity.

**Reply by alrz** [reply]
>case _ when ..., but it doesn't seem to be accepted.

IIRC that is by design. but I don't know why.

**Reply by PathogenDavid** [reply]
@alrz I could only find it mentioned in the LDM notes for [2017-11-29](https://github.com/dotnet/csharplang/blob/addae801a336e473451deb2d95b7de3795d46aae/meetings/2017/LDM-2017-11-29.md#decisions) and [2019-03-13](https://github.com/dotnet/csharplang/blob/addae801a336e473451deb2d95b7de3795d46aae/meetings/2019/LDM-2019-03-13.md#where-to-produce-warnings-for-null-test-in-switch). In both cases though the subject at hand is switch expressions rather than pattern matching as an alternative syntax to what we ended up with.

**Comment by Terry1511512** [comment]
To me it seems like a glaring hole in the switch statement if a "case" can have a "with" expression but a "default" cannot.

**Comment by CyrusNajmabadi** [comment]
I moved this to csharplang, as this is a discussion about a proposed change to the language.

**Comment by HaloFour** [comment]
Has this discussion been narrowed down to the desired syntax?  Supporting a `when` clause with the `default` case, or supporting `_` as a case?  Or both?  Technically `case _` is already legal if `_` is a constant in scope, although the compiler seems to warn in that case now.
